lib/ansible/modules/cloud/azure/azure_rm_servicebus_sas_policy.py

```diff
@@ -0,0 +1,264 @@
+from ansible.module_utils._text import to_native
+from datetime import datetime, timedelta
+
+
+class AzureRMServiceBusSASPolicy(AzureRMModuleBase):
+
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            resource_group=dict(type='str', required=True),
+            name=dict(type='str', required=True),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            namespace=dict(type='str', required=True),
+            queue=dict(type='str'),
+            topic=dict(type='str'),
+            regenerate_primary_key=dict(type='bool'),
+            regenerate_secondary_key=dict(type='bool'),
+            rights=dict(type='str', choices=['manage', 'listen', 'send', 'listen_send'])
+        )
+
+        mutually_exclusive = [
+            ['queue', 'topic']
+        ]
+
+        self.resource_group = None
+        self.name = None
+        self.state = None
+        self.namespace = None
+        self.queue = None
+        self.topic = None
+        self.regenerate_primary_key = None
+        self.regenerate_secondary_key = None
+        self.rights = None
+
+        self.results = dict(
+            changed=False,
+            id=None
+        )
+
+        super(AzureRMServiceBusSASPolicy, self).__init__(self.module_arg_spec,
+                                                         mutually_exclusive=mutually_exclusive,
+                                                         supports_check_mode=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in list(self.module_arg_spec.keys()):
+            setattr(self, key, kwargs[key])
+
+        changed = False
+
+        policy = self.get_auth_rule()
+        if self.state == 'present':
+            if not policy:  # Create a new one
+                changed = True
+                if not self.check_mode:
+                    policy = self.create_sas_policy()
+            else:
+                changed = changed | self.regenerate_primary_key | self.regenerate_secondary_key
+                if self.regenerate_primary_key and not self.check_mode:
+                    self.regenerate_sas_key('primary')
+                if self.regenerate_secondary_key and not self.check_mode:
+                    self.regenerate_sas_key('secondary')
+            self.results = self.policy_to_dict(policy)
+            self.results['keys'] = self.get_sas_key()
+        elif policy:
+            changed = True
+            if not self.check_mode:
+                self.delete_sas_policy()
+
+        self.results['changed'] = changed
+        return self.results
+
+    def _get_client(self):
+        if self.queue:
+            return self.servicebus_client.queues
+        elif self.topic:
+            return self.servicebus_client.topics
+        return self.servicebus_client.namespaces
+
+    # SAS policy
+    def create_sas_policy(self):
+        if self.rights == 'listen_send':
```

Comment:
why not just let the user pass this as a list with `choices` instead of making compound values?